openfisca_core/simulations.py

```diff
@@ -244,6 +244,17 @@ def get_entity(self, entity_type = None, plural = None):
         if plural:
             return [entity for entity in self.entities.values() if entity.plural == plural][0]
 
+    def get_memory_usage(self, variables = None):
+        result = dict(
+            total_nb_bytes = 0,
+            by_variable = {}
+            )
+        for entity in self.entities.itervalues():
+            entity_memory_usage = entity.get_memory_usage(variables = variables)
```

Comment:
The algorithm could be improved performance-wise, I agree. If this function was in the "reactor core" and called each time a variable was computed, it would definitely make sense to optimize it.

As of today, this function is though used in interactive mode for investigation, or in tests scripts. The tiny performance difference would probably not bring much, so I'm not sure whether it's worth the complexity of optimization. What do you think ?